components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/DefaultIdentityProviderImpl.java

```diff
@@ -108,6 +108,36 @@ public String getIdOfUser(String userName) throws IdentityProviderException {
     }
 
     @Override
+    public String getEmailOfUser(String userId) throws IdentityProviderException {
+      Response userResponse = scimServiceStub.getUser(userId);
+        String userEmail;
+        if (userResponse == null) {
+            String errorMessage =
+                    "Error occurred while retrieving Id of user " + userId + ". Error : Response is null.";
+            throw new IdentityProviderException(errorMessage, ExceptionCodes.RESOURCE_RETRIEVAL_FAILED);
```

Comment:
do we need to log this exception? If this doesn't use in rest api layer we have to  log it

Reply:
Fixed with PR: https://github.com/wso2/carbon-apimgt/pull/4537